### DISCUSSION TITLE
Dialog: close on click outside of component

### DISCUSSION THREAD

**Discussion by lfaudreejr** [tailwindlabs/headlessui#2286] [Ideas]
Some of us would like to disable this for clicking outside the component while keeping the functionality for close on escape keypresses. There's already a close event emitted, that could be useful for this scenario if the event was passed back to the user. Would you be willing to add this so we can determine on how and when we want to close the dialog?

**Comment by dungle-scrubs** [comment]
I would also like this feature 👍

**Comment by JohnCampionJr** [comment]
Even just a parameter to the close emit indicating whether it was a click or escape would be helpful.
